thirdeye/thirdeye-pinot/src/main/resources/validators/detection/detection-config-schema.json

```diff
@@ -0,0 +1,214 @@
+      "type": "string"
+    },
+    "active": {
+      "type": "boolean"
+    },
+    "cron": {
+      "type": "string"
+    },
+    "filters": {
+      "$ref": "#/definitions/filters"
+    },
+    "dimensionExploration": {
+      "$ref": "#/definitions/dimensionExploration"
+    },
+    "rules": {
+      "$ref": "#/definitions/rules"
+    },
+    "alerts": {
+      "$ref": "#/definitions/alerts"
+    },
+    "merger": {
+      "$ref": "#/definitions/merger"
+    },
+    "grouper": {
+      "$ref": "#/definitions/grouper"
+    }
+  },
+
+  "definitions": {
+    "alerts": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/subEntity"
+      }
+    },
+
+    "subEntity": {
+      "type": "object",
+      "description": "a sub-entity detection config",
+      "required": [ "name", "type" ],
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "description": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string",
+          "description": "detection pipeline type",
+          "enum": [
+            "COMPOSITE_ALERT",
+            "METRIC_ALERT"
+          ]
+        },
+        "alerts": {
+          "$ref": "#/definitions/alerts"
+        },
+        "metric": {
+          "type": "string"
+        },
+        "dataset": {
+          "type": "string"
+        },
+        "filters": {
+          "$ref": "#/definitions/filters"
+        },
+        "dimensionExploration": {
+          "$ref": "#/definitions/dimensionExploration"
+        },
+        "rules": {
+          "$ref": "#/definitions/rules"
+        },
+        "merger": {
+          "$ref": "#/definitions/merger"
+        },
+        "grouper": {
+          "$ref": "#/definitions/grouper"
+        }
+      },
+      "additionalProperties": false
+    },
+
+    "dimensionExploration": {
+      "type": "object",
+      "description": "exploration combination of dimension values",
+      "properties": {
+        "dimensions": {
+          "type": "array",
+          "description": "dimension column names",
+          "items": {
+            "type": "string"
+          }
+        },
+        "dimensionFilterMetric": {
+          "type": "string"
+        }
+      },
+      "patternProperties": {
+        "k|minValue|minContribution|minValueHourly|maxValueHourly|minValueDaily|maxValueDaily": {
+          "type": "number"
+        }
+      },
+      "additionalProperties": false
+    },
+
+    "filters": {
+      "type": "object",
+      "description": "dimension filters",
+      "patternProperties": {
+        ".*": {
+          "type": "array",
+          "description": "dimension filter values",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+
+    "eachComponent": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "required": [ "type", "name" ],
+        "properties": {
+          "type": {
+            "type": "string"
```

Comment:
shall the rule type be an enum?

Reply:
We can define an enum, but I avoided it for a couple of reasons.
1. We do not want to hard code and maintain the list of possible rule types here.
2. We have internal algorithm types which shouldn't be hardcoded here.
3. We run a validation in the code which will anyways capture if the type is invalid.